plugins/pluginDependencies.js

```diff
@@ -0,0 +1,266 @@
+    }
+    options.discoveryStrategy = options.discoveryStrategy || "disableChildren";
+    return options;
+}
+
+/**
+ * Extends the dependencies graph with either ancestors or descendants of the given plugin
+ * 
+ * @param {object} _graph Dependencies graph 
+ * @param {String} code Plugin id
+ * @param {String} direction Relative direction (up|down)
+ * @returns {undefined} nothing
+ */
+function discoverRelativePlugins(_graph, code, direction) {
+
+    if (!Object.prototype.hasOwnProperty.call(_graph, code)) {
+        return [];
+    }
+
+    direction = direction || "up";
+
+    if (Object.prototype.hasOwnProperty.call(_graph[code], direction)) {
+        return _graph[code][direction];
+    }
+
+    var queue = [code],
+        visited = {},
+        relativeType = "parents";
+    if (direction !== "up") {
+        relativeType = "children";
+    }
+    while (queue.length > 0) {
+        var current = queue.pop();
+        if (Object.prototype.hasOwnProperty.call(visited, current)) {
+            continue;
+        }
+
+        if (!_graph[current]) {
+            visited[current] = 1;
+            continue;
+        }
+
+        for (var item in _graph[current][relativeType]) {
+            queue.push(item);
+        }
+        visited[current] = 1;
+    }
+    var relatives = [];
+
+    delete visited[CLY_ROOT];
+
+    for (var itemCode in visited) {
+        if (itemCode !== code) {
+            relatives.push(itemCode);
+        }
+    }
+
+    _graph[code][direction] = relatives;
+}
+
+/**
+ * Returns a dependency graph of the provided plugins list.
+ * 
+ * If the discovery strategy is "disableChildren", then the returned graph will be a subset of the original dependencies. 
+ * 
+ * For instance,
+ * Let A <- B (plugin B depends on Plugin A);
+ * If A is disabled, A won't be included in the graph, but only B will be. This will cause an unmet dependency error,
+ * and reported via the errors object.
+ * 
+ * In "enableParents" case, however, Plugin A (parent) will be enabled, so the graph will be complete. 
+ * 
+ * This function may report different types of errors; which are generally caused by cyclic dependencies, absent plugins, etc.
+ * 
+ * @param {Array} plugins We use this array of plugin names to create dependency graph.  
+ * @param {object} options Options object
+ * @returns {Object} Returns an object with dpcs and errors fields
+ */
+function getDependencies(plugins, options) {
+    var errors = {},
+        dpcs = {
+            [CLY_ROOT]: {
+                children: {}
+            }
+        };
+
+    var {logger, discoveryStrategy} = getOptions(options);
+
+    if (["disableChildren", "enableParents"].indexOf(discoveryStrategy) === -1) {
+        logger.e(`Invalid discoveryStrategy (${discoveryStrategy}) for analyzeAndFixDependencies.`);
+        return null;
+    }
+
+    var _loadPluginMeta = function(pName) {
+        try {
+            var pkgInfo = require("./" + pName + "/package.json");
```

Comment:
Codacy found an issue: [Found non-literal argument in require](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6259262)